README.md

```diff
@@ -1,91 +1,551 @@
+<p></p>
+The pgai Docker container has all the software you need preinstalled. To connect to
+pgai running in a Docker container:
+
+
+1. In Terminal, navigate to the folder you cloned pgai to. 
+
+1. Build the Docker image:
+
+   ```bash
+   docker build -t pgai .
+   ```
+
+1. Run the container:
+
+    ```bash
+    docker run -d --name pgai -p 9876:5432 -e POSTGRES_PASSWORD=pgaipass pgai
+    ```
+
+1. Connect to the database:
+
+    ```bash
+    psql -d "postgres://postgres:pgaipass@localhost:9876/postgres"
+    ```
+
+1. Create the pgai extension:
+
+    ```sql
+    CREATE EXTENSION ai CASCADE;
+    ```
+   The `CASCADE` automatically installs the plpython3u and pgvector dependencies.
+
+</details>
+<details>
+<summary>Setup a virtual developer environment locally</summary>
+<p></p>
+Best practice is to setup your developer environment in Docker. However, to install pgai in a virtual
+Ubunto environment.
+
+In this repository, [vm.sh](./vm.sh) creates a [multipass](https://multipass.run/) virtual machine called `pgai`. This script 
+installs the [pgai Prerequisites](#pgai-prerequisites) in the `pgai` Ubuntu virtual
+machine. This repo is mounted to `/pgai` in the virtual machine.
+
+1. To create the virtual machine, run the following command:
+
+    ```bash
+    ./vm.sh
+    ```
+
+    You are automatically logged into Terminal on the virtual machine.
+
+1. In the multipass shell, [Setup a developer environment locally](#setup-a-developer-environment-locally).
+
+For more information on using Multipass, [see the documentation](https://multipass.run/docs/use-an-instance).
+
+</details>
+<details>
+<summary>Setup a developer environment locally</summary>
+<a id="setup-a-developer-environment-locally"></a>
+<p></p>
+Best practice is to setup your developer environment in Docker. However, to install pgai directly on your 
+developer environment. 
+
+1. On the command line, navigate to the folder you cloned pgai to.
+
+1. Build pgai:
+
+    ```bash
+    make install
+    ```
+
+    This installs the pgvector and pgai extensions on your local PostgreSQL developer
+    environment, then installs the package dependencies in your Python environment.
+
+   
+1. Create the pgai extension in a database. Use either:
+
+   - SQL:
+      1. Connect to PostgreSQL:
+         ```bash
+         psql -d "postgres://postgres:pgaipass@localhost:9876/postgres"
+         ```
+         TODO: Is this true for the local installation? Is the database already created?
+
+      1. Create the pgai extension:
+   
+          ```sql
+          CREATE EXTENSION IF NOT EXISTS ai CASCADE;
+          ```
+    
+          The `CASCADE` automatically installs the plpython3u and pgvector dependencies.
+
+   - Terminal: 
+     1. In `Makefile`, update `DB` and `USER` to match your PostgreSQL configuration. 
+     1.  Create the pgai extension:
+
+        ```bash
+        make create_extension
+        ```
+</details>
+
+
+## Securely connect to OpenAI through pgai
```

Comment:
```suggestion
## Handling API keys securely
```
We plan to quickly add support for providers other than OpenAI. So, better to keep this generic.